server/src/main/java/org/elasticsearch/index/mapper/IndexFieldMapper.java

```diff
@@ -71,6 +71,11 @@ public Query existsQuery(QueryShardContext context) {
         public IndexFieldData.Builder fielddataBuilder(String fullyQualifiedIndexName, Supplier<SearchLookup> searchLookup) {
             return new ConstantIndexFieldData.Builder(mapperService -> fullyQualifiedIndexName, name(), CoreValuesSourceType.BYTES);
         }
+
+        @Override
+        public ValueFetcher valueFetcher(MapperService mapperService, SearchLookup searchLookup, String format) {
+            return lookup -> Collections.singletonList(mapperService.index().getName());
+        }
```

Comment:
It was, I implemented it because it can be implemented but as you point out later on we don't actually try and fetch values from metadata fields so it should go back to throwing a UOE

Reply:
sounds good thanks, maybe we should re-discuss whether fetching metadata fields should be possible.

Reply:
We're tracking the meta field question in #60985, it would be great to discuss and maybe implement this in a follow-up. 

My intuition is that users will want to fetch metadata fields. One example is `_size`, whose values aren't easy to pull off the response.